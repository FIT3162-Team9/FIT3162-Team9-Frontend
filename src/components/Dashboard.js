import React, { Component, useState, useEffect, useCallback} from 'react'
import clsx from 'clsx';
import { makeStyles, formatMs } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import Items from './NavbarItem';
import './../fonts/Quicksand-Regular.ttf';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import Home from '../pages/Home';
import Bushfire from '../pages/Bushfire';
import Temperature from '../pages/Temperature';
import About from '../pages/About';
import "./../App.css";

import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import Button from '@material-ui/core/Button';

import { getStates, getLGAs, getStations } from "./firebase";

const drawerWidth = 220;

const useStyles = makeStyles((theme) => ({
  root: {
    fontFamily: 'Quicksand',
    display: 'flex',
  },
  toolbar: {
    background: "linear-gradient(to right,rgba(7,71,67,.7),#40C9BF )", //"linear-gradient(to right, #40C9BF, rgba(7,180,67,.3))"
    paddingRight: 24, // keep right padding when drawer closed
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    overflowX: 'hidden',
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  title: {
    color: 'white',//'#255C59',
    fontFamily: 'Quicksand',
    paddingRight: '1450px',
    flexGrow: 1,
  },
  drawerPaper: {
    background: "linear-gradient(  rgb(32,150,160) ,rgba(7,71,67,.5), rgba(7,71,67,.5), rgb(64,201,191,.2))",//"linear-gradient( #40C9BF, rgba(7,71,67,.5),rgba(7,71,67,.37),rgb(64,201,191,.2))",
    position: 'relative',
    whiteSpace: 'nowrap',
    overflowX: 'hidden',
    width: drawerWidth ,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: 'hidden',
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing(7),
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing(9),
    },
  },
  appBarSpacer: theme.mixins.toolbar,
  content: {
    fontFamily: 'Quicksand',
    backgroundColor:
      theme.palette.type === 'light'
        ? theme.palette.grey[100]
        : theme.palette.grey[900],
    flexGrow: 1,
    height: '100vh',
    overflow: 'auto',
  },
  appName:{
    fontFamily: 'Quicksand',
    fontSize: '28px',
    transform: '[{rotate: -90deg}]',
    color: 'white',//'#255C59',
    textAlign: 'center',
    paddingTop: '16px',
  },
  mainItems:{
    padding: '60px 8px',
    textAlign: 'center',
    color: 'white',
  },

}));

export default function Dashboard() {
  const classes = useStyles();
  const [open, setOpen] = React.useState(true);
  const [states, setStates] = React.useState([]);
  const [LGAs, setLGAs] = React.useState([]);
  const [state, setState] = React.useState();
  const [LGA, setLGA] = React.useState();
  const [stations, setStations]  = React.useState();
  const [station, setStation] = React.useState('76031');
  
  
  useEffect(() => {
  }, [LGA]);

  const handleSelectedPage = () => {
    console.log("works");
   };

   React.useEffect(() => {
    const states = getStates();
    console.log('states', setStates(states));
   }, [])

  return (
    <Router>
    <div className={classes.root}>
      <CssBaseline />
      <AppBar
        position="absolute"
        className={clsx(classes.appBar, open && classes.appBarShift)}
      >
        <Toolbar className={classes.toolbar}>
          <Typography
            component="h1"
            variant="h5"
            color="inherit"
            noWrap
            className={classes.title}
          >
            <Switch>
              <Route exact path ="/">DASHBOARD</Route>
              <Route exact path ="/temperature">VISUALISATION </Route>
              <Route exact path ="/bushfire">ANALYSIS</Route>
              <Route exact path ="/about">ABOUT</Route>
            </Switch>
          </Typography>
        </Toolbar>
      </AppBar>
        <Drawer
          variant="permanent"
          classes={{
            paper: clsx(classes.drawerPaper, !open && classes.drawerPaperClose),
          }}
          open={open}
        >
        <Typography
              component="h1"
              variant="h5"
              color="white"
              className={classes.appName}
            >
              S P â–³ R K
            </Typography>
            <Typography style={{marginTop: 20, color: 'white'}} component="h2" color="white" variant="h6">Station ID: {station}</Typography>
            <div style={{marginTop: 10}}>
              {states !== [] && 
              <FormControl>
                <InputLabel id="state-label">State</InputLabel>
                <Select
                  labelId="state-label"
                  id="state-select"
                  value={states[0]}
                  onChange={(val)=>{console.log(val.target.value); setState(val.target.value); getLGAs(val.target.value, setLGAs);}}
                  style={{width: 160}}
                >
                  {states.map((val, index) => <MenuItem key={index} value={val}>{val.toUpperCase()}</MenuItem>)}
                </Select>
              </FormControl>
              }
            </div>
            <div style={{marginTop: 10}}>
              {(LGAs !== [] && state) &&
              <FormControl>
                <InputLabel id="lga-label">LGAs</InputLabel>
                <Select
                  labelId="lga-label"
                  id="lga-select"
                  value={LGAs[0]}
                  onChange={(val)=>{
                    console.log(val.target.value);
                    setLGA(val.target.value);
                    getStations(state, val.target.value, 
                      (stationVal) => {
                        setStation(stationVal[0]);
                        setStations(stationVal);
                      });
                  }}
                  style={{width: 160}}
                >
                  {LGAs.map((val, index) => <MenuItem key={index} value={val}>{val.split("_").join(" ").toUpperCase()}</MenuItem>)}
                </Select>
              </FormControl>
              }
            </div>
            <div style={{marginTop: 10}}>
              {(LGAs !== [] && state && station && stations) &&
              <FormControl>
                <InputLabel id="stations-label">Stations</InputLabel>
                <Select
                  labelId="stations-label"
                  id="stations-select"
                  value={station}
                  onChange={(val)=>{
                    console.log(val.target.value);
                    setStation(val.target.value);
                  }}
                  style={{width: 160}}
                >
                  {stations.map((val, index) => <MenuItem key={index} value={val}>{val}</MenuItem>)}
                </Select>
              </FormControl>
              }
            </div>
            {/* <Button 
              onClick={console.log}
              variant="contained"
              disabled={!station}
              style={{marginTop: 20, width: 160, alignSelf: 'center'}}
            >Set Station</Button> */}
            
          <List className={classes.mainItems}> <Items action={handleSelectedPage} /> </List>
        </Drawer>
        <main className={classes.content}>
          <div className={classes.appBarSpacer} />
            <Switch>
              <Route exact path ="/" render={() => <Home station={station} LGA={LGA}/>}/>
              <Route exact path ="/bushfire" render={() => <Bushfire station={station} LGA={LGA}/>}/>
              <Route exact path ="/temperature" render={() => <Temperature station={station}/>} />
              <Route exact path ="/about" component={About}/>
            </Switch>
        </main>
    </div>
    </Router>
    
  );
}
