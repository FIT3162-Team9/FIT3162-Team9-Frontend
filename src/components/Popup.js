import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Slide from '@material-ui/core/Slide';

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

export default function Popup(props) {
  
  const prop = props.props
  return (
    <div>
      <Dialog
        open={prop.open}
        TransitionComponent={Transition}
        keepMounted
        onClose={prop.handleClose}
        aria-labelledby="alert-dialog-slide-title"
        aria-describedby="alert-dialog-slide-description"
      >
        <DialogTitle id="alert-dialog-slide-title">{"Data Unavailable"}</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-slide-description">
            We currently only support the following LGAs in Victoria:
            <ul>
              <li>Mildura Shire</li>
              <li>Glenelg</li>
              <li>Greater Bendigo</li>
              <li>Ballarat North</li>
              <li>Wellington</li>
              <li>Horsham</li>
              <li>Banyule</li>
              <li>Brimbank</li>
            </ul>
            The time ranges also start at 1st January 2009, except for Horsham which starts at 1st January 2013.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={prop.handleClose} color="primary">
            Okay
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}