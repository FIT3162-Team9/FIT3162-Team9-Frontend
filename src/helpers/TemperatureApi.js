import * as firebase from "firebase/app";
import moment from 'moment';

import { firebaseConfig } from "./firebase";

import "firebase/analytics";
import "firebase/auth";
import "firebase/firestore";

// Initialize Firebase if not already initialized
if (!firebase.apps.length) {
  firebase.initializeApp(firebaseConfig);
}

// Tests the temperature api
export const testing = async () => {
  const db = firebase.firestore();

  var tempStationRef = db.collection('data').doc('temperature').collection('76031').orderBy("timestamp").startAt(moment(1568657396).unix()).endAt(moment(1569506400).unix());

  tempStationRef.onSnapshot((snapshot) => {
    const tempData = [12];
    snapshot.forEach((doc) => tempData.push({ ...doc.data(), id: doc.id }));
    return tempData
  });

}

/**
 * return historic temperature data based on LGA and date range
 * @param   {number}    station            Station number
 * @param   {function}  setTempData        Function to set temperature array from the module that called it
 * @param   {number}    startAt            Timestamp at the start date
 * @param   {number}    endAt              Timestamp at the end date
 */  
export const getTemperature = async (station='', setTempData, startAt, endAt) => {
  const db = firebase.firestore();

  var tempStationRef = db.collection('data').doc('temperature').collection(station).orderBy("timestamp").startAt(startAt).endAt(endAt);
  
  
  return tempStationRef.onSnapshot((snapshot) => {
    const tempData = [];
    snapshot.forEach((doc) => tempData.push({ ...doc.data(), id: doc.id }));
    console.log('Fetched all data', tempData);
    setTempData(tempData);
  });
}

/**
 * return forecasted temperature data based on LGA and date range
 * @param   {number}    station            Station number
 * @param   {function}  setTempData        Function to set temperature data from the module that called it
 * @param   {number}    startAt            Timestamp at the start date
 * @param   {number}    endAt              Timestamp at the end date
 * @return  {array}                        Array of forecasted temperature values by timestamp 
 */  
export const getForecastedTemperature= async (station='', setTempData, startAt, endAt) => {
  const db = firebase.firestore();
  var tempStationRef = db.collection('forecast').doc('temperature').collection(station).orderBy("timestamp").startAt(startAt).endAt(endAt);
  
  return tempStationRef.onSnapshot((snapshot) => {
    let tempData = [];
    snapshot.forEach((doc) => tempData.push({ ...doc.data(), id: doc.id }));
    console.log('Fetched all data', tempData);
    setTempData(tempData);
  });
}