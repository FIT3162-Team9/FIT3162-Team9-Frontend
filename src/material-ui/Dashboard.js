import * as React from 'react';
import clsx from 'clsx';
import { makeStyles, formatMs } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import Drawer from '@material-ui/core/Drawer';
import Box from '@material-ui/core/Box';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import Badge from '@material-ui/core/Badge';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Link from '@material-ui/core/Link';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import NotificationsIcon from '@material-ui/icons/Notifications';
import { mainListItems, secondaryListItems } from './listItems';
import Items from './listItems';
import Chart from './Chart';
import './../fonts/Quicksand-Regular.ttf';
import {BrowserRouter as Router, Route, Switch, NavLink} from 'react-router-dom';
import home from './../pages/home';
import Bushfire from './../pages/bushfire';
import Temperature from '../pages/Temperature';
import About from './../pages/about';
import "./../App.css";


//Base Template from Material-UI
//https://github.com/mui-org/material-ui/tree/master/docs/src/pages/getting-started/templates

function Copyright() {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Copyright © '}
      <Link color="inherit" href="https://material-ui.com/">
        Your Website
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const drawerWidth = 220;

const useStyles = makeStyles((theme) => ({
  root: {
    fontFamily: 'Quicksand',
    display: 'flex',
  },
  toolbar: {
    background: "linear-gradient(to right,rgba(7,71,67,.7),#40C9BF )", //"linear-gradient(to right, #40C9BF, rgba(7,180,67,.3))"
    paddingRight: 24, // keep right padding when drawer closed
  },
  toolbarIcon: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    ...theme.mixins.toolbar,
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    overflowX: 'hidden',
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: 36,
  },
  menuButtonHidden: {
    display: 'none',
  },
  title: {
    color: 'white',//'#255C59',
    fontFamily: 'Quicksand',
    paddingRight: '1450px',
    flexGrow: 1,
  },
  drawerPaper: {
    background: "linear-gradient(  rgb(32,150,160) ,rgb(7,71,67,.6), rgb(7,71,67,.6), rgb(7,71,67,.6), rgba(7,71,67,.6), rgba(7,71,67,.6), rgba(7,71,67,.5), rgba(7,71,67,.5), rgb(64,201,191,.2))",//"linear-gradient( #40C9BF, rgba(7,71,67,.5),rgba(7,71,67,.37),rgb(64,201,191,.2))",
    position: 'relative',
    whiteSpace: 'nowrap',
    overflowX: 'hidden',
    width: drawerWidth ,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: 'hidden',
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing(7),
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing(9),
    },
  },
  appBarSpacer: theme.mixins.toolbar,
  content: {
    fontFamily: 'Quicksand',
    backgroundColor:
      theme.palette.type === 'light'
        ? theme.palette.grey[100]
        : theme.palette.grey[900],
    flexGrow: 1,
    height: '100vh',
    overflow: 'auto',
  },
  container: {
    paddingTop: theme.spacing(4),
    paddingBottom: theme.spacing(4),
  },
  paper: {
    padding: theme.spacing(2),
    display: 'flex',
    overflow: 'auto',
    flexDirection: 'column',
  },
  fixedHeight: {
    height: 240,
  },
  appName:{
    fontFamily: 'Quicksand',
    fontSize: '28px',
    transform: '[{rotate: -90deg}]',
    color: 'white',//'#255C59',
    textAlign: 'center',
    paddingTop: '16px',
  },
  mainItems:{
    padding: '60px 8px',
    textAlign: 'center',
    color: 'white',
  },
  test:{
     fontSize: '100px',
  }
}));

export default function Dashboard() {
  const classes = useStyles();
  const [open, setOpen] = React.useState(true);
  
  const handleSelectedPage = () => {
    console.log("works");
   };
  const handleDrawerOpen = () => {
    setOpen(true);
  };
  const handleDrawerClose = () => {
    setOpen(false);
  };
  
  const fixedHeightPaper = clsx(classes.paper, classes.fixedHeight);

  return (
    <Router>
    <div className={classes.root}>
      <CssBaseline />
      <AppBar
        position="absolute"
        className={clsx(classes.appBar, open && classes.appBarShift)}
      >
        <Toolbar className={classes.toolbar}>
          {/* <IconButton
            edge="start"
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            className={clsx(
              classes.menuButton,
              open && classes.menuButtonHidden,
            )}
          >
            <MenuIcon />
          </IconButton> */}
          <Typography
            component="h1"
            variant="h5"
            color="inherit"
            noWrap
            className={classes.title}
          >
            <Switch>
              <Route exact path ="/">DASHBOARD</Route>
              <Route exact path ="/temperature">VISUALISATION </Route>
              <Route exact path ="/bushfire">ANALYSIS</Route>
              <Route exact path ="/about">ABOUT</Route>
            </Switch>
          </Typography>
        </Toolbar>
      </AppBar>
        <Drawer
          variant="permanent"
          classes={{
            paper: clsx(classes.drawerPaper, !open && classes.drawerPaperClose),
          }}
          open={open}
        >
        <Typography
              component="h1"
              variant="h5"
              color="white"
              className={classes.appName}
            >
              S P △ R K
            </Typography>
          {/* <div className={classes.toolbarIcon}>
            <IconButton onClick={handleDrawerClose}>
              <ChevronLeftIcon />
            </IconButton>
          </div> */}
          <List className={classes.mainItems}> <Items action={handleSelectedPage} /> </List>
          {/* <List>{secondaryListItems}</List> */}
        </Drawer>
        <main className={classes.content}>
          <div className={classes.appBarSpacer} />
            <Switch>
              <Route exact path ="/" component={home}/>
              <Route exact path ="/bushfire" component={Bushfire}/>
              <Route exact path ="/temperature" component={Temperature}/>
              <Route exact path ="/about" component={About}/>
            </Switch>
          {/* <Container maxWidth="lg" className={classes.container}>
            <Grid container spacing={3}>
              Chart 
              <Grid item xs={12} md={8} lg={9}>
                <Paper className={fixedHeightPaper}>
                  <Chart />
                </Paper>
              </Grid>
              Recent Deposits
              <Grid item xs={12} md={4} lg={3}>
                <Paper className={fixedHeightPaper}>
                  <Deposits />
                </Paper>
              </Grid>
              Recent Orders 
              <Grid item xs={12}>
                <Paper className={classes.paper}>
                  <Orders />
                </Paper>
              </Grid>
            </Grid>
            <Box pt={4}>
              <Copyright />
            </Box>
          </Container> */}
        </main>
    </div>
    </Router>
    
  );
}
