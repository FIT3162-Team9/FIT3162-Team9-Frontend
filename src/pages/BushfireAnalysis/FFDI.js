

/**
     * return forecasted bushfire ratings based on climate input
     * @param   {number} temp               Temperature value of date X
     * @param   {object} climate            props with key value {Humidity,Windspeed}
     * @param   {number} climate.humidity   Humidity value of date X
     * @param   {number} climate.windspeed  Windspeed value of date X
     * @return  {number}                    Forecasted Bushfire Rating
     */  

    export const FFDI = (temp, climate, droughtFactor) => {
        let defaultHumidity = 40;
        let defaultWindspeed = 40;
        let constant = -0.45;
        let drought = 0.987 * Math.log(parseInt(droughtFactor ? droughtFactor : 0));
        let humidity = climate ? 0.0345 * climate['humidity'] : 0.0345 * (parseInt(defaultHumidity));
        let temperature = 0.0338 * (temp);
        let windspeed = climate ? 0.0234 * climate['windspeed'] : 0.0234 * (parseInt(defaultWindspeed));
        let exponential = (constant + drought - humidity + temperature + windspeed)
        let ratings = 2 * Math.exp(exponential)
        
        
        if (ratings<1){
          return 1
        }
        else if (ratings>150) {
          return 150
        }
        else{
          return Math.floor(ratings * 100) / 100
        }
      }