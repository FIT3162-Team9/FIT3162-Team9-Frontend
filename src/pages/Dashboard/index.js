import React, { Component, useState, useEffect } from 'react'
import PropTypes from 'prop-types';

import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import clsx from 'clsx'
import { makeStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import Button from '@material-ui/core/Button'
import Circle from './Circle'
import moment from 'moment'
import Divider from '@material-ui/core/Divider'
import DateRangePicker from '@wojtekmaj/react-daterange-picker';

import { getTemperature, getForecastedTemperature } from '../../helpers/TemperatureApi';
import { getHumidityWind } from '../../helpers/HumidityWindApi';
import { snakeToTitle } from '../../helpers/utils'

const useStyles = makeStyles(theme => ({
    root: {
        fontFamily: 'Quicksand',
    },
    paper: {      
      overflow: 'auto',
      flexGrow: 1,
      height: 'auto',
      width: 'auto',
      padding: '15px',
      paddingTop: '5px',
      margin: '15px',
    },
    container: {
      paddingTop: theme.spacing(0),
      paddingBottom: theme.spacing(4),
    },
    title: {
      color: 'black',//'#255C59',
      fontFamily: 'Quicksand',
      paddingRight: '1450px',
      paddingLeft: '480px',
      paddingTop: '-200px',
      flexGrow: 1,
      fontSize: '40px',
      
    },
    typography: {
      marginLeft: '20px',
      marginTop: '20px',
      fontSize: '20px',
      display: 'flex',
    },
    circle: {
      fontSize: '50px',
    },
    paperColor: {
      background: "white" ,
    },
    outerPaper: {
      display: 'flex',
      flexGrow: 1,
      flexDirection: 'column',
      alignItems: 'center',
      margin: 'auto',
      padding: '20px',
      width: '80%',
      background: "transparent",
      boxShadow: 'none',
    },
    headerFont:{
      fontSize: '20px',
      opacity: '1',
    },
    dateFont:{
      fontSize: '15px',
      opacity: '1',
    },
    addButton:{
      background: '#316e80',
      fontSize: '12px',
      margin: '20px 20px 25px 20px',
      color: 'white',
      opacity: '100%',
    },
    clearButton: {
      fontSize: '12px',
      margin: '0px 0px 5px 0px',
      opacity: '100%',
    }
  }))

  /**
 * Home component page to be displayed when selected on navigation bar.
 * Computes Bushfire Ratings using FFDI Model and climate data called from firebase.
 * Display forecasted Bushfire Risk to assist users with their travel planning
 * @param {object} props                props object with key value {station,LGA}
 * @param {string} props.station  Station ID selected by user
 * @param {string} props.LGA     Local Government Area selected by user
 */

function Home(props) {
    const classes = useStyles()
    const autoHeightColoredPaper = clsx(classes.paper, classes.paperColor)
    const [userLog, setUserLog] = useState([]);
    const [dateRange, setDateRange] = useState([moment().subtract(13, 'months'), moment().subtract(1, 'month')]);
    const LGA = props ? props.LGA : undefined
    const station = props? props.station: undefined
    const [tempData , setTempData] = useState();
    const [pastTempData , setPastTempData] = useState();
    const [humidityWindData, setClimateData] = useState();
   
    /*FFDI Ratings Calculation*/

  
    const FFDI = (temp, climate) =>{
      let constant = -0.45;
      let drought = 0.987 * Math.log(10);
      let humidity = climate ? 0.0345 * climate['humidity'] : 0.0345 * (5);
      let temperature = 0.0338 * (temp);
      let windspeed = climate ? 0.0234 * climate['windspeed'] : 0.0234 * (80);
      let exponential = (constant + drought - humidity + temperature + windspeed)
      let ratings = 2 * Math.exp(exponential)
      console.log("ffdi",temp,climate['humidity'],climate['windspeed'])
      console.log(ratings)
      
      if (ratings<1){
        return 1
      }
      else if (ratings>150) {
        return 150
      }
      else{
        return ratings
      }
    }

    const callData = () => {
        if (!LGA || !station) {return}
        let formattedDateRange = dateRange.map(date => moment(date).unix());
        const startTimestamp = formattedDateRange[0];
        const endTimestamp = formattedDateRange[1]
        getForecastedTemperature(station,setTempData,startTimestamp,endTimestamp)
        getTemperature(station,setPastTempData,startTimestamp,endTimestamp)
        getHumidityWind(LGA,setClimateData,startTimestamp,endTimestamp)
    }

    /*FFDI Ratings Calculation*/
    const addLog = () => {
      if (!LGA) {return} 
      if (!tempData) { return }
      let empty = [];
      console.log('data', humidityWindData, pastTempData, tempData);

      var dict = new Map();
      humidityWindData.forEach((doc) => dict.set(doc['timestamp'],{humidity:doc['humidity'],windspeed:doc['windspeed']}));
      pastTempData.forEach((doc) => empty.push({day:doc['day'],month:doc['month'],humidity:dict.get(doc['timestamp']) ? dict.get(doc['timestamp'])['humidity'] : null ,
                                                windspeed:dict.get(doc['timestamp']) ? dict.get(doc['timestamp'])['windspeed'] : null,
                                                max:doc['max'],bushfirerating:FFDI(doc['max'],dict.get(doc['timestamp']))}));
      tempData.forEach((doc) => empty.push({day:doc['day'],month:doc['month'],humidity:dict.get(doc['timestamp']) ? dict.get(doc['timestamp'])['humidity'] : null,
                                            windspeed:dict.get(doc['timestamp']) ? dict.get(doc['timestamp'])['windspeed'] : null,
                                            max:doc['max'],bushfirerating:FFDI(doc['max'],dict.get(doc['timestamp']))}));
        
        let tempUserLog = [userLog];
        let circleList = [];
        empty.forEach((doc) => circleList.push(<Circle className={classes.circle} day = {doc['day']} month = {doc['month']} bushfirerating = {doc['bushfirerating']} props={doc} />))
        tempUserLog.push(<Grid item xs={12} md={8} lg={9}>
                          <Paper className={autoHeightColoredPaper}>
                            <Typography className={classes.typography}>
                              {LGA ? snakeToTitle(LGA) : "No LGA Selected"} 
                            </Typography>
                            {circleList}
                          </Paper>
                          <Divider/>
                        </Grid>
                          );
        setUserLog(tempUserLog);
       
    } 

    const clearPage = () => {
      setUserLog([])
    }

    useEffect(() => {
      addLog();
    }, [humidityWindData]);

    useEffect(() => {
    }, [userLog])

    return (
        <Container maxWidth="lg" className={classes.container}>
            <Grid container spacing={3}>
              <Paper className={classes.outerPaper}>
                  <Grid item xs={12} md={8} lg={9}>
                    <Typography className={classes.headerFont}>
                        Selected LGA: {LGA ? snakeToTitle(LGA) : "None"}
                    </Typography>
                    <Typography className={classes.headerFont}>
                        Selected Station: {station ? station : "None"}
                    </Typography>
                    <DateRangePicker className={classes.dateFont}
                      onChange={setDateRange}
                      value={dateRange}
                      />
                    <Button className={classes.addButton} onClick={()=>callData()}>
                        Add User Log
                    </Button>
                    <Button onClick={()=>clearPage()}>
                      Clear
                    </Button>
                    <Divider></Divider>
                </Grid>            
                {userLog}
              </Paper>
            </Grid>
        </Container>
    )
    }

Home.propTypes = {
  /**
   * LGA selected by user
   * To perform API call to firebase to fetch humidity/windspeed data
   */
  LGA: PropTypes.string.isRequired,
  /**
   * Station ID seleceted by user 
   * To perform API call to firebase to fetch temperature data
   */
  station: PropTypes.string.isRequired,
  
};

export default Home
