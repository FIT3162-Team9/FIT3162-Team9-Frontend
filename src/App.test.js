import React from 'react';
import { render } from '@testing-library/react';

import { shallow, mount } from "enzyme";
import { act } from 'react-dom/test-utils';
import toJson from "enzyme-to-json";
import About from './pages/About'
import {FFDI} from './pages/BushfireAnalysis/FFDI';
import Home from './pages/Dashboard';
import ShallowRenderer from 'react-test-renderer/shallow';
import CircularProgressWithLabel,{bushfireColor} from './pages/Dashboard/Circle';
import Dashboard, {Routes} from './pages';
import { MemoryRouter } from 'react-router';
import { Route } from 'react-router-dom';
import Temperature from './pages/TemperatureAnalysis';
import {testing} from './helpers/TemperatureApi';
import Popup from './components/Popup';

it("About page renders without crashing", () => {
  shallow(<About />);
});

it("About page renders correctly", () => {
  const tree = shallow(<About />);
  expect(toJson(tree)).toMatchSnapshot();
});


describe('Test for FFDI', () => {
  it('should return correct bushfire rating', () => {
    expect(FFDI(60,{humidity:100,windspeed:200},10)).toBe(150);
    
  });
});

describe("<Home />", () => {
  it("Renders <Home /> component correctly", () => {

    const {getByText}= render(<Home LGA='Banyule' station='76031' />);
    expect(getByText(/Banyule/i)).toBeInTheDocument();
  });
});

describe('Circle display correct data', () => {

  it('Date Displayed correctly', () =>{
    const {getByText} = render(<CircularProgressWithLabel day={17} month={3} bushfirerating={19.97}/>);
    expect(getByText('17/3')).toBeInTheDocument();  
  });

  it('Date Displayed correctly', () =>{
    const {getByText} = render(<CircularProgressWithLabel day={17} month={3} bushfirerating={19.97}/>);
    expect(getByText('17/3')).toBeInTheDocument();  
  });

  it('Color returned correctly', () =>{
    expect(bushfireColor(80)).toBe('#ff4040');
  });
});

describe('FFDI correct data', () => {
  it('Test case 1', () => {
     expect(FFDI(40,{humidity:40,windspeed:40},5)).toBe(15.48)
  });

  it('Test case 2', () => {
    expect(FFDI(40,{humidity:50,windspeed:80},8)).toBe(44.45)
 });

});

// describe('routes using memory router', () => {

//   it('should show Home component for / router (using memory router)', () => {
//     const component = mount(<MemoryRouter initialentries="{['/about]}" initialIndex={0}>
//         <Routes/>
//      </MemoryRouter>
//     );
//     expect(component.find(<About/>)).toHaveLength(1);
//   })


// })

// it("navigates home when you click the logo", async => {
//   // in a real test a renderer like "@testing-library/react"
//   // would take care of setting up the DOM elements
//   const root = document.createElement('div');
//   document.body.appendChild(root);

//   // Render app
//   render(
//     <MemoryRouter initialEntries={['/my/initial/route']}>
//       <App />
//     </MemoryRouter>,
//     root
//   );

//   // Interact with page
//   act(() => {
//     // Find the link (perhaps using the text content)
//     const goHomeLink = document.querySelector('#nav-logo-home');
//     // Click it
//     goHomeLink.dispatchEvent(new MouseEvent("click", { bubbles: true }));
//   });

//   // Check correct page content showed up
//   expect(document.body.textContent).toBe('Home');
// });

// describe('api call', () => {
//   it('success', () =>{
//     expect(testing()).toBe([])
//   } )
// }) 

