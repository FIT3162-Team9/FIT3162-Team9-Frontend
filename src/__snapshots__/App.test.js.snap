// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page renders correctly 1`] = `
<div>
  Modify content in pages folder
</div>
`;
